### ISSUE TITLE
`jx shell` not working with zsh

### ISSUE THREAD

**Issue by rawlingsj** [jenkins-x/jx#1702]
```
Creating a new shell using the kubernetes context foo
Bash RC file is /tmp/jx-shell-rcfile-246678195

All changes to the kuberentes context like changing environment, namespace or context will be local to this shell
To return to the global context use the command: exit

zsh:1: /tmp/jx-shell-rcfile-246678195: restricted
error: exit status 1
```

**Comment by johnmcollier** [comment]
I've verified this too with zsh.

I've also verified that `jx shell` doesn't work with the following other shells:
- tcsh
- csh
- ksh

**Comment by beebird** [comment]
I'm facing the same issue, just created a PR #2032 to fix this.

**Comment by ccojocar** [comment]
Fixed by PR #2032.
